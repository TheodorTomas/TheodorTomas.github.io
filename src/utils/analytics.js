import * as ReactGA from 'react-ga';

// Click to action analytics utils
export const CTAAnalytics = ({ category, ...options }) => ReactGA.event({ action: 'Click', category, ...options });
