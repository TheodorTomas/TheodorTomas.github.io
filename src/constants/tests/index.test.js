import analytics from '../analytics';

describe('constants/analytics', () => {
  const OLD_ENV = process.env;

  beforeEach(() => {
    jest.resetModules();
    process.env = { ...OLD_ENV };
    delete process.env.NODE_ENV;
  });

  afterEach(() => {
    process.env = OLD_ENV;
  });

  it('should render matching snapshot when not in production', () => {
    expect(analytics).toMatchSnapshot();
  });

  it('should render matching snapshot when in production', () => {
    process.env.NODE_ENV = 'production';
    const productionAnalytics = require('../analytics').default; // eslint-disable-line global-require

    expect(productionAnalytics).toMatchSnapshot();
  });
});
