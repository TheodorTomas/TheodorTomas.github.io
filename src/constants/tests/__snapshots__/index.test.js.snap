// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constants/analytics should render matching snapshot when in production 1`] = `
Object {
  "trackerId": "UA-109205585-1",
}
`;

exports[`constants/analytics should render matching snapshot when not in production 1`] = `
Object {
  "trackerId": "UA-000000000-0",
}
`;
